### DISCUSSION TITLE
Countdown circle timer is working even user leaves the app

### DISCUSSION THREAD

**Discussion by dorapax-lab** [vydimitrov/react-countdown-circle-timer#129] [Q&A] [answered]
Hi, I'm using Countdown circle timer for my app. I want to stop the countdown when user leaves my app (not kill, just leave) and go to any other app. But Countdown circle timer is not getting stopped and countdown is working in background. This will affect user's device battery. Is there a way to stop countdown when user leaves the app ?

Thanks.

**Comment by vydimitrov** [comment, marked as answer]
Hey, check the [AppState](https://reactnative.dev/docs/appstate#app-states), when it goes to `background` just set `isPlaying={false}`

**Reply by dorapax-lab** [reply]
Thank you so much! @vydimitrov 
Countdown circle timer is one of the cute component I've seen. Thank you so much for your contribution.

**Reply by vydimitrov** [reply]
Sure thing, just for reference. *To my knowledge*: the Countdown is using `Animated` component, which internally is using `requestAnimationFrame`, so when the app goes in the background the `requestAnimationFrame` calls will be suspended until the moment the app comes back focused. However, the time will continue meanwhile, so when you come back you will see that the time in the countdown changed but that does not mean that it was playing on the background. This is also how it works on the Web.
